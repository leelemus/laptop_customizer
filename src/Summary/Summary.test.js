import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Summary from './Summary';


describe(`Summary Component`, () => {

  const selected = {
    Processor: {
        name: '17th Generation Intel Core HB (7 Core with donut spare)',
        cost: 700
      },
    "Operating System": {
        name: 'Ubuntu Linux 16.04',
        cost: 200
      },
    "Video Card":{
        name: 'Toyota Corolla 1.5v',
        cost: 1150.98
      },
    Display: {
        name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
        cost: 1500
      }
  };

  it('renders Summary with data', () => {
    const wrapper = 
        shallow(<Summary
                    selected={selected}
                />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
