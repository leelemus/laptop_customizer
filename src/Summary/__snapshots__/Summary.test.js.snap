// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary Component renders Summary with data 1`] = `
<section
  className="main__summary"
>
  <h3>
    NEW GREENLEAF 2018
  </h3>
  <div
    className="summary__option"
    key="Processor"
  >
    <div
      className="summary__option__label"
    >
      Processor
        
    </div>
    <div
      className="summary__option__value"
    >
      17th Generation Intel Core HB (7 Core with donut spare)
    </div>
    <div
      className="summary__option__cost"
    >
      $700.00
    </div>
  </div>
  <div
    className="summary__option"
    key="Operating System"
  >
    <div
      className="summary__option__label"
    >
      Operating System
        
    </div>
    <div
      className="summary__option__value"
    >
      Ubuntu Linux 16.04
    </div>
    <div
      className="summary__option__cost"
    >
      $200.00
    </div>
  </div>
  <div
    className="summary__option"
    key="Video Card"
  >
    <div
      className="summary__option__label"
    >
      Video Card
        
    </div>
    <div
      className="summary__option__value"
    >
      Toyota Corolla 1.5v
    </div>
    <div
      className="summary__option__cost"
    >
      $1,150.98
    </div>
  </div>
  <div
    className="summary__option"
    key="Display"
  >
    <div
      className="summary__option__label"
    >
      Display
        
    </div>
    <div
      className="summary__option__value"
    >
      15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs
    </div>
    <div
      className="summary__option__cost"
    >
      $1,500.00
    </div>
  </div>
  <Total
    selected={
      Object {
        "Display": Object {
          "cost": 1500,
          "name": "15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs",
        },
        "Operating System": Object {
          "cost": 200,
          "name": "Ubuntu Linux 16.04",
        },
        "Processor": Object {
          "cost": 700,
          "name": "17th Generation Intel Core HB (7 Core with donut spare)",
        },
        "Video Card": Object {
          "cost": 1150.98,
          "name": "Toyota Corolla 1.5v",
        },
      }
    }
  />
</section>
`;
